src/edu/csus/ecs/pc2/ui/ShadowCompareRunsPane.java

```diff
@@ -327,11 +339,114 @@ private ShadowCompareRunsSummaryPane getSummaryPanel() {
         
     }
     
+    /**
+     * Initialized dynamically refresh panel that contains checkbox,textfield and label for auto refreshing.
+     * @return
+     */
+    private JPanel getdynamicallyRefreshPanel() {
+        
+        if (dynamicallyRefreshPanel == null) {
+            dynamicallyRefreshPanel = new JPanel();
+            
+            JCheckBox checkbox = new JCheckBox("Automatically Refresh Every:");
+            dynamicallyRefreshPanel.add(checkbox);
+            
+            
+            JTextField textField = new JTextField("5",2);
+            ((AbstractDocument) textField.getDocument()).setDocumentFilter(new DocumentFilter() { //Makes the textfield so that user is not allowed to enter illegal numbers.
+                @Override
+                public void replace(FilterBypass fb, int offset, int length, String text, AttributeSet attrs) throws BadLocationException {
+                    String currentText = fb.getDocument().getText(0, fb.getDocument().getLength());
+                    StringBuilder newText = new StringBuilder(currentText);
+                    newText.replace(offset, offset + length, text);
+                    
+                    try {
+                        int value = Integer.parseInt(newText.toString());
+                        if (value >= 1 && value <= 60) {
+                            // Allow values between 1 and 60
+                            super.replace(fb, offset, length, text, attrs);
+                        } else {
+                            JOptionPane.showMessageDialog(null, "Duration of automatic refresh has to be between 1 and 60 seconds", "Warning", JOptionPane.WARNING_MESSAGE);
+                            Toolkit.getDefaultToolkit().beep();
+                        }
+                    } catch (NumberFormatException e) {
+                        Toolkit.getDefaultToolkit().beep();
+                    }
+                    
+                }
+            });
+            
+            dynamicallyRefreshPanel.add(textField);
+            
+            checkbox.addItemListener(new ItemListener() {
+                Timer timer;
+                @Override
+                public void itemStateChanged(ItemEvent e) {
+                    
+                    
+                    if (e.getStateChange() == ItemEvent.SELECTED) {
+                        if (textField.getText().isEmpty()) {
+                            JOptionPane.showMessageDialog(null, "Please enter a value to textField", "Error", JOptionPane.ERROR_MESSAGE);
+                            return;
+                        }
+                        int time;
+                        try {
+                            time = Integer.parseInt(textField.getText());
+                            if (time < 1) {
+                                log.log(Log.WARNING, "dynamicallyRefreshPanel received time less than 1"+ textField.getText());
```

Comment:
I'm thinking you need a space between the string and the "textField.getText()"; eg. ... received time less than 1 "+textField.getText()   (Note you need a space after the "1" and before the double-quote.)  This happens in a few of the messages below which I will not comment on. But all should be fixed.